### ISSUE TITLE
Universal Search

### ISSUE THREAD

**Issue by evrowe** [healthsparq/ember-fountainhead#40]
Dependent on #39

Once the base search functionality is complete, we want to build a universal-search available throughout the consuming app, to be triggered by a set of hotkeys. The interface would be something similar to spotlight in macOS or using cmd/ctrl+p in Atom, and would allow for a developer to quickly search through an app's documentation and open the doc pages directly without having to first visit the `/api` route.

This functionality should be mostly able to leverage the work being done to make searching available in the docs themselves. Criteria specific to this feature are:

- [ ] Custom search interface in consuming app
- [ ] Custom keyboard shortcut trigger (re-bindable via config?)
- [ ] Automatic injection of FH services (already done) and components 🤔 into the consuming app so that this can actually display; hopefully the components can be done automatically but it may require the user to drop a quick component reference into their `application.hbs` template
- [ ] Optional: quick-preview mode for docs within app context? Might not be practical without slight reworking of how docs components and data fetching is handled
